spec/factorial_specs.rb

```diff
@@ -0,0 +1,50 @@
+require 'rspec'
+require_relative '../solver'
+
+describe Solver do
+  describe '#factorial' do
+    context 'when number is 0' do
+      it 'returns 1' do
+        expect(Solver.factorial(0)).to eq(1)
+      end
+    end
+
+    context 'when number is 1' do
+      it 'returns 1' do
+        expect(Solver.factorial(1)).to eq(1)
+      end
+    end
+
+    context 'when number is valid' do
+      it 'returns 2' do
+        expect(Solver.factorial(2)).to eq(2)
+      end
+
+      it 'returns 6' do
+        expect(Solver.factorial(3)).to eq(6)
+      end
+
+      it 'returns 24' do
+        expect(Solver.factorial(4)).to eq(24)
+      end
+
+      it 'returns 120' do
+        expect(Solver.factorial(5)).to eq(120)
+      end
+
+      it 'returns 720' do
+        expect(Solver.factorial(6)).to eq(720)
+      end
+    end
+
+    context 'when number is invalid' do
+      it 'should raise an error' do
+        expect { Solver.factorial(-1) }.to raise_error(ArgumentError, 'n must be non-negative')
+      end
+
+      it 'should raise an error' do
+        expect { Solver.factorial('a') }.to raise_error(ArgumentError, 'n must be non-negative')
```

Comment:
- [ ] Good job implementing the changes required by the previous reviewer 👏  But some tests are still failing as you can see on the screenshot below ⬇️ . Please make to fix the errors and make all your tests pass as required for this project 👍 

![factorials_test](https://user-images.githubusercontent.com/6615394/229291467-bbad677e-3e80-4321-bf8a-e75fc954153c.PNG)

Reply:
![test](https://user-images.githubusercontent.com/108669691/229295409-4434eef6-9ccd-4f8b-b7d3-da5260a80c0f.PNG)
Test is passing please approve